p2p/pex/addrbook.go

```diff
@@ -726,6 +648,44 @@ func (a *addrBook) addAddress(addr, src *p2p.NetAddress) error {
 	return nil
 }
 
+func (a *addrBook) randomPickAddresses(bucketType byte, num int) []*p2p.NetAddress {
+	var buckets []map[string]*knownAddress
+	switch bucketType {
+	case bucketTypeNew:
+		buckets = a.bucketsNew
+	case bucketTypeOld:
+		buckets = a.bucketsOld
+	default:
+		panic("unexpected bucketType")
+	}
+	total := 0
+	for _, bucket := range buckets {
+		total = total + len(bucket)
+	}
+	addresses := make([]*knownAddress, 0, total)
+	for _, bucket := range buckets {
+		for _, ka := range bucket {
+			addresses = append(addresses, ka)
+		}
+	}
+	selection := make([]*p2p.NetAddress, 0, num)
+	chosenSet := make(map[string]bool, num)
+	rand.Shuffle(total, func(i, j int) {
```

Comment:
Just noting that this is slightly more deterministic now than previously. I do not think this is a problem (doesn't look like it was) but previously the common package randomness was used which seeds the global randomness using the crypto/rand package while math/rand is seeded with `1`: https://golang.org/src/math/rand/rand.go?#L288
To get a behaviour more like the previous one could call `rand.Seed` with some int derived from crypto/rand